cassandra/cluster.py

```diff
@@ -1772,8 +1989,31 @@ def _create_response_future(self, query, parameters, trace, custom_payload, time
         elif isinstance(query, PreparedStatement):
             query = query.bind(parameters)
 
-        cl = query.consistency_level if query.consistency_level is not None else self.default_consistency_level
-        serial_cl = query.serial_consistency_level if query.serial_consistency_level is not None else self.default_serial_consistency_level
+        if self.cluster._config_mode == _ConfigMode.LEGACY:
+            if execution_profile is not EXEC_PROFILE_DEFAULT:
+                raise ValueError("Cannot specify execution_profile while using legacy parameters.")
+
+            if timeout is _NOT_SET:
+                timeout = self.default_timeout
+
+            cl = query.consistency_level if query.consistency_level is not None else self.default_consistency_level
+            serial_cl = query.serial_consistency_level if query.serial_consistency_level is not None else self.default_serial_consistency_level
+
+            retry_policy = query.retry_policy or self.cluster.default_retry_policy
+            row_factory = self.row_factory
+            load_balancing_policy = self.cluster.load_balancing_policy
+        else:
+            execution_profile = self._get_execution_profile(execution_profile)
```

Comment:
This implies that the execution_profile could be a string (e.g. the name of a profile) or an ExecutionProfile instance. If the latter, isn't it possible that the ExecutionProfile is not configured with available hosts, not registered in the profile-manager, etc. Wouldn't it be safer to require the arg to be a registered profile name only?